pkg/controller/machine/controller.go

```diff
@@ -344,6 +346,12 @@ func (r *ReconcileMachine) drainNode(machine *machinev1.Machine) error {
 		DryRun: false,
 	}
 
+	for _, condition := range node.Status.Conditions {
```

Comment:
can this be a function e.g isNodeUnreachable with a unit test?

Reply:
👍